impl/src/test/java/org/ehcache/docs/Ehcache2.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright Terracotta, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.ehcache.docs;
+
+import net.sf.ehcache.*;
+import net.sf.ehcache.config.CacheConfiguration;
+import org.junit.Test;
+
+
+public class Ehcache2 {
+
+  @Test
+  // tag::CustomExpiryEhcache2[]
+  public void ehcache2Expiry() throws Exception {
+    CacheManager cacheManager = initCacheManager();
+    CacheConfiguration cacheConfiguration = new CacheConfiguration().name("cache")
+        .maxEntriesLocalHeap(100)
+        .timeToLiveSeconds(20);   // <1>
+    cacheManager.addCache(new Cache(cacheConfiguration));
+
+    Element element = new Element(10L, "Hello");
+
+    // Set the expiry of the element
+    element.setTimeToLive(getTimeToLiveInSeconds((Long)element.getObjectKey(), (String)element.getObjectValue()));  // <2>
```

Comment:
I would make that example a bit more evolved, getting the TTL and only if different than default above set it.